### ISSUE TITLE
Rules disappear in private browsing

### ISSUE THREAD

**Issue by grenzor** [FirefoxBar/HeaderEditor#30]
When you use a private browsing window all the rules disappear. Also it's not possible to import rules as after you choose your backup file nothing happens.

Using Firefox latest stable.

P.S. Probably unrelated to the above but from time to time all my rules (even in normal browsing window) disappear. I have no idea how to reproduce this issue but it has happened to me 3-4 times so far so the problem looks like it is persistent and not a one-off.

**Comment by sylingd** [comment]
It's Firefox's BUG
You can try this method: [https://github.com/FirefoxBar/xStyle/wiki/DB-Error](https://github.com/FirefoxBar/xStyle/wiki/DB-Error)

**Comment by grenzor** [comment]
I tried both methods but still having the same problem. 

Have you filed the bug on Bugzilla? Or is there already any existing open issue?

**Comment by sylingd** [comment]
[https://github.com/openstyles/stylus/pull/170](https://github.com/openstyles/stylus/pull/170)
[https://bugzilla.mozilla.org/show_bug.cgi?id=1389951](https://bugzilla.mozilla.org/show_bug.cgi?id=1389951)

**Comment by grenzor** [comment]
@sylingd I found the reason why rules disappeared (wasn't related to Private browsing in my case). I found that some cleaner programs will delete the header rules/DB if Firefox storage is set to be cleaned. All my settings for other addons are still there even when Firefox storage is set to be cleaned only HE rules get deleted, so I'm guessing it is stored differently than the other addons. 

Also just some info `browser.storage.local` API in Firefox seems to have some issues that Chrome doesn't have: https://bugzilla.mozilla.org/show_bug.cgi?id=1371255
